libcloud/compute/drivers/scaleway.py

```diff
@@ -0,0 +1,615 @@
+    {
+        'id': 'X64-30GB',
+        'name': 'X64-30GB',
+        'ram': 30720,
+        'disk': 300,
+        'bandwidth': 500,
+        'price': 0.1,
+        'extra': {
+            'cores': 8,
+            'monthly': 49.99,
+            'range': 'Intensive',
+            'arch': 'x86_64',
+        },
+    },
+    {
+        'id': 'X64-60GB',
+        'name': 'X64-60GB',
+        'ram': 61440,
+        'disk': 700,  # TODO: Update to _minimum_ required storage
+        'bandwidth': 1000,
+        'price': 0.18,
+        'extra': {
+            'cores': 10,
+            'monthly': 89.99,
+            'range': 'Intensive',
+            'arch': 'x86_64',
+        },
+    },
+    {
+        'id': 'X64-120GB',
+        'name': 'X64-120GB',
+        'ram': 122880,
+        'disk': 1000,  # TODO: Update to _minimum_ required storage
+        'bandwidth': 1000,
+        'price': 0.36,
+        'extra': {
+            'cores': 12,
+            'monthly': 179.99,
+            'range': 'Intensive',
+            'arch': 'x86_64',
+        },
+    },
+]
+
+# The API also doesn't give location info, so we provide it ourselves, instead.
+SCALEWAY_LOCATION_DATA = [
+    {'id': 'par1', 'name': 'Paris 1', 'country': 'FR'},
+    {'id': 'ams1', 'name': 'Amsterdam 1', 'country': 'NL'},
+]
+
+
+class ScalewayResponse(JsonResponse):
+    valid_response_codes = [httplib.OK, httplib.ACCEPTED,
+                            httplib.CREATED, httplib.NO_CONTENT]
+
+    def parse_error(self):
+        return super(ScalewayResponse, self).parse_error()['message']
+
+    def success(self):
+        return self.status in self.valid_response_codes
+
+
+class ScalewayConnection(ConnectionUserAndKey):
+    """
+    Connection class for the Scaleway driver.
+    """
+
+    host = SCALEWAY_API_HOSTS['default']
+    allow_insecure = False
+    responseCls = ScalewayResponse
+
+    def request(self, action, params=None, data=None, headers=None,
+                method='GET', raw=False, stream=False, region=None):
+        if region:
+            old_host = self.host
+            self.host = SCALEWAY_API_HOSTS[region.id
+                                           if isinstance(region, NodeLocation)
+                                           else region]
+            if not self.host == old_host:
+                self.connect()
+
+        return super(ScalewayConnection, self).request(action, params, data,
+                                                       headers, method, raw,
+                                                       stream)
+
+    def add_default_headers(self, headers):
+        """
+        Add headers that are necessary for every request
+        """
+        headers['X-Auth-Token'] = self.key
+        headers['Content-Type'] = 'application/json'
+        return headers
+
+
+def _to_lib_size(size):
+    return int(size / 1000 / 1000 / 1000)
+
+
+def _to_api_size(size):
+    return int(size * 1000 * 1000 * 1000)
+
+
+class ScalewayNodeDriver(NodeDriver):
```

Comment:
This is part of the public API, all methods need to be documented, see https://libcloud.readthedocs.io/en/latest/development.html#docstring-conventions and included in the documentation, see https://github.com/apache/libcloud/blob/trunk/libcloud/compute/drivers/gce.py#L1673 for example.

Reply:
I'll handle this some time this week. Sorry for the delay!

Reply:
Fixed in 29fe802.